From ce09480b36d1ca656e9b5f308fc3e2762860bd3f Mon Sep 17 00:00:00 2001
From: Jaroslav Kysela <jkysela@redhat.com>
Date: Wed, 3 Sep 2014 07:12:53 -0400
Subject: [alsa] hda: Correct AD1986A 3stack pin configs

Message-id: <140972823573-20-patchset-jkysela@redhat.com>
Patchwork-id: 90720
O-Subject: [RHEL 7.1 PATCH 019/442] ALSA: hda - Correct AD1986A 3stack pin configs
Bugzilla: 1112200
RH-Acked-by: John Linville <linville@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>

The 3stack pin configs for AD1986A codec had incorrect values that
resulted in broken mic and line-in.

Bugzilla: https://bugzilla.kernel.org/show_bug.cgi?id=66621
Cc: <stable@vger.kernel.org>
Signed-off-by: Takashi Iwai <tiwai@suse.de>

(picked from kernel: ed0e0d0617a8dc3d8b82c6e54827f269f2247b07)
rhbz#1112200
Signed-off-by: Jaroslav Kysela <jkysela@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/sound/pci/hda/patch_analog.c b/sound/pci/hda/patch_analog.c
index b174eb1..819d4a9 100644
--- a/sound/pci/hda/patch_analog.c
+++ b/sound/pci/hda/patch_analog.c
@@ -281,11 +281,11 @@ static const struct hda_fixup ad1986a_fixups[] = {
   .v.pins = (const struct hda_pintbl[]) {
    { 0x1a, 0x02214021 }, /* headphone */
    { 0x1b, 0x01014011 }, /* front */
-   { 0x1c, 0x01013012 }, /* surround */
-   { 0x1d, 0x01019015 }, /* clfe */
+   { 0x1c, 0x01813030 }, /* line-in */
+   { 0x1d, 0x01a19020 }, /* rear mic */
    { 0x1e, 0x411111f0 }, /* N/A */
    { 0x1f, 0x02a190f0 }, /* mic */
-   { 0x20, 0x018130f0 }, /* line-in */
+   { 0x20, 0x411111f0 }, /* N/A */
    {}
   },
  },
-- 
1.7.1